src/Schema/Directives/BaseDirective.php

```diff
@@ -292,4 +301,101 @@ protected function loadArgValues(): void
             $this->directiveArgs[$node->name->value] = AST::valueFromASTUntyped($node->value);
         }
     }
+
+    /**
+     * Get an instance of a directive.
+     *
+     * @api
+     */
+    public static function getDirectiveInstance(string $directiveSource, ScalarTypeDefinitionNode|ScalarTypeExtensionNode|ObjectTypeDefinitionNode|ObjectTypeExtensionNode|InterfaceTypeDefinitionNode|InterfaceTypeExtensionNode|UnionTypeDefinitionNode|UnionTypeExtensionNode|EnumTypeDefinitionNode|EnumTypeExtensionNode|InputObjectTypeDefinitionNode|InputObjectTypeExtensionNode|FieldDefinitionNode|InputValueDefinitionNode|EnumValueDefinitionNode $node): Directive
```

Comment:
I don't like the name. `get...` indicates to me that this only retrieves something that was already there, and certainly not mutate its input. Why not `addDirectiveToNode` as proposed?

Also, this function and the other helpers should better go into `ASTHelper`. Given their static nature, nothing really binds them to `BaseDirective`.